wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/DefaultServicesStartStrategy.java

```diff
@@ -67,32 +67,20 @@
                                                                               service.getVolumesFrom().size());
 
             for (String dependsOn : service.getDependsOn()) {
-                if (!services.containsKey(dependsOn)) {
-                    throw new IllegalArgumentException(
-                            format("Dependency '%s' in machine '%s' points to not known machine.",
-                                   dependsOn, serviceEntry.getKey()));
-                }
+                checkDependency(dependsOn, serviceEntry.getKey(), services, "A service can not depend on itself");
```

Comment:
We don't have 'service' entity in our public API. User would not understand what that mean if environment is not docker compose.
And message doesn't point to a machine with the error.